interactions/client/bot.py

```diff
@@ -363,6 +365,55 @@ async def change_presence(self, presence: ClientPresence) -> None:
         """
         await self._websocket._update_presence(presence)
 
+    async def get(
+        self, model: T, http_method: Optional[str] = None, **kwargs: Dict[str, Snowflake]
+    ) -> T:
+        r"""
+        A helper function for getting a model from the Discord API.
+
+        :param model: The model to get.
+        :type model: object
+        :param http_method: The HTTP method to use instead of `get_x`.
+        :type http_method: Optional[str]
+        :param \**kwargs: The arguments to pass to the HTTP method.
+        :type \**kwargs: Dict[str, Snowflake]
+        :return: The model.
+        :rtype: object
+        """
+        if model is Role:
+            if not kwargs.get("guild_id") and kwargs.get("role_id"):
+                raise TypeError(
+                    "Client.get() missing some of the following required keyword arguments: guild_id, role_id."
+                )
+            try:
+                roles: List[dict] = await self._http.get_all_roles(int(kwargs["guild_id"]))
+            except TypeError as e:
+                raise ValueError("Invalid guild_id provided.") from e
+            role = next((_role for _role in roles if _role["id"] == str(kwargs["role_id"])), None)
+            if role is None:
+                raise ValueError("Role not found.")
+            return Role(**role, _client=self._http)
+        else:
+            class_name: str = model.__name__.removesuffix("s")
+            try:
+                _http_method: Coroutine = getattr(
+                    self._http, http_method or f"get_{class_name.lower()}"
+                )
+            except AttributeError as e:
+                raise ValueError(f"Client.get() does not support the model {class_name}.") from e
+            try:
+                res: Union[dict, List[dict]] = await _http_method(**kwargs)
+            except TypeError as e:
+                raise ValueError(
+                    f"Client.get() could not find a {class_name} with the given IDs."
+                ) from e
+            if isinstance(res, dict):
+                return model(**res, _client=self._http)
+            elif isinstance(res, list):
+                return [model(**r, _client=self._http) for r in res]
```

Comment:
This should be consistent with http requests too. If you're allowing a list of results to be produced, a list of requests should be callable as well. In this case, the code needs to check for if a "list"-like http endpoint method exists, otherwise to set up an async queue.

Reply:
The code does not repeatedly request, it just requests once.
The only reason it is there is because of the `http_method` argument, which lets you put a string, such as `"get_self_guilds"` or other methods, which *returns* a list of results.

Reply:
This comment refers to https://github.com/interactions-py/library/pull/702#pullrequestreview-931088201 and https://github.com/interactions-py/library/pull/702#discussion_r842230296

Reply:
Oh, I see.

Reply:
> such as "get_self_guilds" or other methods, which returns a list of results.

You're not actually utilizing the http method feeder variable in this method impl itself to check for a suffix of "s," in-fact, you're actually removing it from the equation. This could be very harmful towards bot developers who want to make only 1 HTTP call instead of many.

Reply:
No, I'm removing the "s" from the model.
Well I can remove that tbh
the only use is for `interactions.Webhooks` lmao

Reply:
There's no reason to be removing anything from the model names unless you're trying to parse a GW event name in its raw form.

Reply:
There is a `get_webhook` but no `get_webhooks`.
The "s" at the end was the problem.
I think I'll just remove the `.removesuffix("x")`.